[2024-02-18T06:05:27.733+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: final_project.Stage_events manual__2024-02-18T06:05:26.075589+00:00 [queued]>
[2024-02-18T06:05:27.737+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: final_project.Stage_events manual__2024-02-18T06:05:26.075589+00:00 [queued]>
[2024-02-18T06:05:27.737+0000] {taskinstance.py:2171} INFO - Starting attempt 1 of 1
[2024-02-18T06:05:27.743+0000] {taskinstance.py:2192} INFO - Executing <Task(StageToRedshiftOperator): Stage_events> on 2024-02-18 06:05:26.075589+00:00
[2024-02-18T06:05:27.746+0000] {standard_task_runner.py:60} INFO - Started process 20543 to run task
[2024-02-18T06:05:27.748+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'final_project', 'Stage_events', 'manual__2024-02-18T06:05:26.075589+00:00', '--job-id', '26', '--raw', '--subdir', 'DAGS_FOLDER/final_project.py', '--cfg-path', '/tmp/tmpbhsav8zs']
[2024-02-18T06:05:27.750+0000] {standard_task_runner.py:88} INFO - Job 26: Subtask Stage_events
[2024-02-18T06:05:27.778+0000] {task_command.py:423} INFO - Running <TaskInstance: final_project.Stage_events manual__2024-02-18T06:05:26.075589+00:00 [running]> on host 6b28116ad14a
[2024-02-18T06:05:27.818+0000] {taskinstance.py:2481} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='udacity' AIRFLOW_CTX_DAG_ID='final_project' AIRFLOW_CTX_TASK_ID='Stage_events' AIRFLOW_CTX_EXECUTION_DATE='2024-02-18T06:05:26.075589+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-02-18T06:05:26.075589+00:00'
[2024-02-18T06:05:27.824+0000] {base.py:83} INFO - Using connection ID 'aws_credentials' for task execution.
[2024-02-18T06:05:27.825+0000] {connection_wrapper.py:378} INFO - AWS Connection (conn_id='aws_credentials', conn_type='aws') credentials retrieved from login and password.
[2024-02-18T06:05:27.835+0000] {base.py:83} INFO - Using connection ID 'redshift' for task execution.
[2024-02-18T06:05:31.894+0000] {sql.py:432} INFO - Running statement: create schema if not exists staging, parameters: None
[2024-02-18T06:05:35.080+0000] {stage_redshift.py:52} INFO - Drop table staging.staging_events from destination Redshift table
[2024-02-18T06:05:35.084+0000] {base.py:83} INFO - Using connection ID 'redshift' for task execution.
[2024-02-18T06:05:38.947+0000] {sql.py:432} INFO - Running statement: Drop table if exists staging.staging_events, parameters: None
[2024-02-18T06:05:41.028+0000] {stage_redshift.py:55} INFO - Create table staging.staging_events from destination Redshift table
[2024-02-18T06:05:41.033+0000] {base.py:83} INFO - Using connection ID 'redshift' for task execution.
[2024-02-18T06:05:45.775+0000] {sql.py:432} INFO - Running statement: 
    create table if not exists staging.staging_events (
        artist text,
        auth text,
        firstname text,
        gender text,
        iteminsession int,
        lastname text,
        length double precision,
        level text,
        location text,
        method text,
        page text,
        registration bigint,
        sessionid int,
        song text,
        status int,
        ts bigint,
        useragent varchar(1000),
        userid text
    )
    , parameters: None
[2024-02-18T06:05:48.733+0000] {stage_redshift.py:58} INFO - Copying data from S3 to Redshift
[2024-02-18T06:05:48.738+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_start' or 'logical_date' instead.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:05:48.740+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'next_ds' from the template is deprecated and will be removed in a future version. Please use '{{ data_interval_end | ds }}' instead.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:05:48.742+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'next_ds_nodash' from the template is deprecated and will be removed in a future version. Please use '{{ data_interval_end | ds_nodash }}' instead.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:05:48.743+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'next_execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_end' instead.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:05:48.743+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'prev_ds' from the template is deprecated and will be removed in a future version.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:05:48.745+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'prev_ds_nodash' from the template is deprecated and will be removed in a future version.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:05:48.745+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'prev_execution_date' from the template is deprecated and will be removed in a future version.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:05:48.746+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'prev_execution_date_success' from the template is deprecated and will be removed in a future version. Please use 'prev_data_interval_start_success' instead.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:05:48.746+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'tomorrow_ds' from the template is deprecated and will be removed in a future version.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:05:48.747+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'tomorrow_ds_nodash' from the template is deprecated and will be removed in a future version.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:05:48.750+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'yesterday_ds' from the template is deprecated and will be removed in a future version.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:05:48.752+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'yesterday_ds_nodash' from the template is deprecated and will be removed in a future version.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:05:48.765+0000] {base.py:83} INFO - Using connection ID 'redshift' for task execution.
[2024-02-18T06:05:53.712+0000] {sql.py:432} INFO - Running statement: 
            COPY staging.staging_events
            FROM 's3://ericliu-udacity-lake-house/log-data'
            ACCESS_KEY_ID 'AKIAXNLLYAMLPRRBUZMF'
            SECRET_ACCESS_KEY '***'
            IGNOREHEADER 1
            DELIMITER ','
        , parameters: None
[2024-02-18T06:06:04.251+0000] {taskinstance.py:2699} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/opt/airflow/plugins/custom_operators/stage_redshift.py", line 71, in execute
    redshift.run(formatted_sql)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 392, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 437, in _run_command
    cur.execute(sql_statement)
psycopg2.errors.InternalError_: Load into table 'staging_events' failed.  Check 'sys_load_error_detail' system table for details.

[2024-02-18T06:06:04.262+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=final_project, task_id=Stage_events, execution_date=20240218T060526, start_date=20240218T060527, end_date=20240218T060604
[2024-02-18T06:06:04.271+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 26 for task Stage_events (Load into table 'staging_events' failed.  Check 'sys_load_error_detail' system table for details.
; 20543)
[2024-02-18T06:06:04.302+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-02-18T06:06:04.312+0000] {taskinstance.py:3281} INFO - 0 downstream tasks scheduled from follow-on schedule check
